tests/gbs/test_similarity.py

```diff
@@ -194,3 +196,91 @@ def test_sample_max_count_deterministic(self, photon_num, count, monkeypatch):
+    print('orbit = ', orbit)
+    print('max_photon = ', max_photon)
+    print('expected = ', expected)
+
+    assert similarity.orbit_cardinality(list(orbit), max_photon) == expected
+
+
+events = [
+    [5, 3, 6, 216],
+    [6, 3, 6, 336],
+    [5, 2, 6, 126],
+    [5, 3, 7, 413],
+    [6, 3, 7, 728],
+    [5, 2, 7, 266],
+]
+
+
+@pytest.mark.parametrize("photons, max_count, modes, expected", events)
+def test_event_cardinality(photons, max_count, modes, expected):
+    """Test if function ``strawberryfields.gbs.similarity.event_cardinality`` returns the
+    correct number of samples for some hard-coded examples."""
+
+    assert similarity.event_cardinality(photons, max_count, modes) == expected
+
+
+class TestProbOrbitMC:
+    """Tests for the function ``strawberryfields.gbs.similarity.prob_orbit_mc.``"""
+
+    def test_mean_computation_orbit(self, monkeypatch):
+        """Tests if the calculation of the sample mean is performed correctly. The test
+        monkeypatches the fock_prob function so that the probability is the same for each sample and
+        is equal to 1/5, i.e., one over the number of samples in the orbit [1,1,1,1] for 5 modes."""
+        graph = nx.complete_graph(5)
+        with monkeypatch.context() as m:
+            m.setattr(
+                "strawberryfields.backends.gaussianbackend.GaussianState.fock_prob",
+                lambda *args, **kwargs: 0.2,
+            )
+
+            assert np.allclose(similarity.p_orbit_mc(graph, [1, 1, 1, 1]), 1.0)
+
+    def test_prob_vacuum_orbit(self):
+        """Tests if the function gives the right probability for the empty orbit when the GBS
+        device has been configured to have zero mean photon number."""
+        graph = nx.complete_graph(10)
+
+        assert similarity.prob_orbit_mc(graph, [], 0) == 1.0
+
+
+class TestProbEventMC:
+    """Tests for the function ``strawberryfields.gbs.similarity.prob_event_mc.``"""
+
+    def test_prob_vacuum_event(self):
+        """Tests if the function gives the right probability for an event with zero photons when
+        the GBS device has been configured to have zero mean photon number."""
+        graph = nx.complete_graph(10)
+
+        assert similarity.prob_event_mc(graph, 0, 0, 0) == 1.0
+
+    def test_mean_event(self, monkeypatch):
+        """Tests if the calculation of the sample mean is performed correctly. The test
+        monkeypatches the fock_prob function so that the probability is the same for each sample and
+        is equal to 1/216, i.e., one over the number of samples in the event with 5 modes,
+        6 photons, and max 3 photons per mode."""
+        graph = nx.complete_graph(6)
+        with monkeypatch.context() as m:
+            m.setattr(
+                "strawberryfields.backends.gaussianbackend.GaussianState.fock_prob",
+                lambda *args, **kwargs: 1.0 / 336,
+            )
+
+            assert np.allclose(similarity.prob_event_mc(graph, 6, 3) - 1.0)
```

Comment:
```suggestion
            assert np.allclose(similarity.prob_event_mc(graph, 6, 3), 1.0)
```

Reply:
Yep this has been done too. thanks Josh!